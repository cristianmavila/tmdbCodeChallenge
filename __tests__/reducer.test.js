import MovieReducer from '../src/app/reducers/reducer_movies'
import {
  FETCH_MOVIES,
  FETCH_START,
  FETCH_SEARCH,
  FETCH_ERROR,
  FETCH_MOVIE,
  initialState } from '../src/app/actions/index';

describe('reducers', () => {

  it('should handle FETCH_MOVIES', () => {
    expect(MovieReducer(initialState, {
      type: FETCH_MOVIES
    })).toMatchSnapshot()
  })

  it('should handle FETCH_MOVIE', () => {
    expect(MovieReducer(initialState, {
      type: FETCH_MOVIE
    })).toMatchSnapshot()
  })

  it('should handle FETCH_SEARCH', () => {
    expect(MovieReducer(initialState, {
      type: FETCH_SEARCH
    })).toMatchSnapshot()
  })

  it('should handle FETCH_START', () => {
    expect(MovieReducer(initialState,{
      type: FETCH_START
    })).toMatchSnapshot()
  })

  it('should handle FETCH_ERROR', () => {
    expect(MovieReducer(initialState,{
      type: FETCH_ERROR
    })).toMatchSnapshot()
  })
})