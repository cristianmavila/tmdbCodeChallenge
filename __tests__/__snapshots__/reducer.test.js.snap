// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers should handle FETCH_ERROR 1`] = `
Object {
  "error": undefined,
  "loading": false,
  "movie": Object {
    "genres": Array [],
  },
  "movies": Object {
    "results": Array [],
  },
}
`;

exports[`reducers should handle FETCH_MOVIE 1`] = `
Object {
  "error": "",
  "loading": false,
  "movie": undefined,
  "movies": Object {
    "results": Array [],
  },
}
`;

exports[`reducers should handle FETCH_MOVIES 1`] = `
Object {
  "error": "",
  "loading": false,
  "movie": Object {
    "genres": Array [],
  },
  "movies": undefined,
}
`;

exports[`reducers should handle FETCH_SEARCH 1`] = `
Object {
  "error": "",
  "loading": false,
  "movie": Object {
    "genres": Array [],
  },
  "movies": undefined,
}
`;

exports[`reducers should handle FETCH_START 1`] = `
Object {
  "error": "",
  "loading": true,
  "movie": Object {},
  "movies": Object {
    "results": Array [],
  },
}
`;
